diff --git a/phpMQTT.php b/phpMQTT.php
index 92b65c6..6ebddeb 100644
--- a/phpMQTT.php
+++ b/phpMQTT.php
@@ -147,7 +147,7 @@ class phpMQTT
             );
             $this->socket = stream_socket_client('tls://' . $this->address . ':' . $this->port, $errno, $errstr, 60, STREAM_CLIENT_CONNECT, $socketContext);
         } else {
-            $this->socket = stream_socket_client('tcp://' . $this->address . ':' . $this->port, $errno, $errstr, 60, STREAM_CLIENT_CONNECT);
+            $this->socket = stream_socket_client('tls://' . $this->address . ':' . $this->port, $errno, $errstr, 60, STREAM_CLIENT_CONNECT);
         }
 
         if (!$this->socket) {
